service/controller/resource/nodepool/delete.go

```diff
@@ -83,9 +100,42 @@ func (r *Resource) removeNodePool(ctx context.Context, azureMachinePool *capzexp
 	return nil
 }
 
+// Deletes all the node objects belonging to the node pool using the k8s API.
+// This happens automatically eventually, but we make this much quicker by doing it on the API server directly.
+func (r *Resource) removeNodesFromK8s(ctx context.Context, ctrlClient client.Client, azureMachinePool *capzexpv1alpha3.AzureMachinePool) error {
+	r.Logger.LogCtx(ctx, "message", fmt.Sprintf("Deleting nodes from k8s API for machine pool %s", azureMachinePool.Name))
+
+	nodeList := &corev1.NodeList{}
+
+	var labelSelector client.MatchingLabels
+	{
+		labelSelector = make(map[string]string)
+		labelSelector[label.MachinePool] = azureMachinePool.Name
+	}
+
+	err := ctrlClient.List(ctx, nodeList, labelSelector)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	r.Logger.LogCtx(ctx, "message", fmt.Sprintf("Found %d nodes to be deleted", len(nodeList.Items)))
+
+	for _, n := range nodeList.Items {
+		r.Logger.LogCtx(ctx, "message", fmt.Sprintf("Deleting node %s", n.Name))
+		err = ctrlClient.Delete(ctx, &n)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
```

Comment:
This can be simplified with `ctrlClient.DeleteAllOf()`. See example here: https://github.com/giantswarm/azure-operator/pull/995/files#diff-c39281c6c70716ac1ac58232d2aa488bR195

Reply:
:exploding_head:

Reply:
done

Reply:
Nice! Fixed